client/client.go

```diff
@@ -97,14 +98,66 @@ func NewClient(cfg *Config) (*Client, error) {
 	return c, nil
 }
 
+type newRequestOpts struct {
+	refPath  string
+	rawQuery string
+	absURI   *url.URL
+	body     io.Reader
+}
+
+type newRequestOptFunc func(*newRequestOpts)
+
+func newRequestOptReferenceURI(path string) newRequestOptFunc {
+	return func(o *newRequestOpts) {
+		o.refPath = path
+	}
+}
+
+func newRequestOptQuery(query string) newRequestOptFunc {
+	return func(o *newRequestOpts) {
+		o.rawQuery = query
+	}
+}
+
+func newRequestOptAbsoluteURI(u *url.URL) newRequestOptFunc {
+	return func(o *newRequestOpts) {
+		o.absURI = u
+	}
+}
+
+func newRequestOptBody(r io.Reader) newRequestOptFunc {
+	return func(o *newRequestOpts) {
+		o.body = r
+	}
+}
+
+func (c *Client) newGETRequest(ctx context.Context, opts ...newRequestOptFunc) (*http.Request, error) {
+	return c.newRequest(ctx, http.MethodGet, opts...)
+}
+
+func (c *Client) newPOSTRequest(ctx context.Context, opts ...newRequestOptFunc) (*http.Request, error) {
+	return c.newRequest(ctx, http.MethodPost, opts...)
+}
+
 // newRequest returns a new Request given a method, path (relative or
 // absolute), rawQuery, and (optional) body.
-func (c *Client) newRequest(method, path, rawQuery string, body io.Reader) (*http.Request, error) {
-	u := c.BaseURL.ResolveReference(&url.URL{
-		Path:     path,
-		RawQuery: rawQuery,
-	})
-	r, err := http.NewRequest(method, u.String(), body)
+func (c *Client) newRequest(ctx context.Context, method string, opts ...newRequestOptFunc) (*http.Request, error) {
```

Comment:
If we're going to diverge further from [http.NewRequestWithContext](https://pkg.go.dev/net/http#NewRequestWithContext), it would be nice to explicitly call out what the options are to set the `path` and `body`.